test/jest.ts

```diff
@@ -0,0 +1,152 @@
+import { describe, expect, test } from '@jest/globals';
+export { expect } from '@jest/globals';
+import { dirname } from 'path';
+import {
+    createProgram,
+    findConfigFile,
+    flattenDiagnosticMessageText,
+    getPreEmitDiagnostics,
+    parseJsonConfigFileContent,
+    readConfigFile,
+    sys
+} from 'typescript';
+
+export type Test = () => void;
+
+export function runTests(
+    canisterName: string,
+    tests: Test,
+    cwd: string = process.cwd()
+) {
+    const { shouldRunTests, shouldRunTypeChecks, shouldRunBenchmark } =
+        processEnvVars();
+    if (shouldRunTests) {
+        describe(`Azle ${canisterName} tests`, tests);
+    }
+    if (shouldRunTypeChecks) {
+        describe(`Azle ${canisterName} type checks`, () => {
+            it('checks types', () => {
+                expect(runTypeCheck(cwd)).toHaveLength(0);
+            });
+        });
+    }
+    if (shouldRunBenchmark) {
+        describe(`Azle ${canisterName} benchmark`, () => {});
+    }
+}
+
+export function wait(name: string, delay: number) {
+    test(
+        `wait ${name}`,
+        async () => {
+            console.info(`Waiting: ${delay} milliseconds ${name}`);
+            await new Promise((resolve) => {
+                setTimeout(resolve, delay);
+            });
+            expect(true);
+        },
+        delay > 2_500 ? delay * 2 : 5_000 // Calling this function takes a few more milliseconds than what is specified in the delay. Since we never want the delay to time out we will add an ample amount of extra time to the timeout. Since doubling a short delay may not give us enough extra time we will have a reasonable floor.
+    );
+}
+
+export function please(
+    name: string,
+    fn: () => void | Promise<void>,
+    timeout?: number
+) {
+    test(
+        `please ${name}`,
+        async () => {
+            console.info(`Preparing: ${name}`);
+            await fn();
+        },
+        timeout
+    );
+}
+
+export function it(
+    name: string,
+    fn: () => void | Promise<void>,
+    timeout?: number
+) {
+    test(
+        `it ${name}`,
+        async () => {
+            console.info(`Testing: ${name}`);
+            await fn();
+        },
+        timeout
+    );
+}
+
+function runTypeCheck(projectDir: string): string[] {
+    const configPath = findConfigFile(
+        projectDir,
+        sys.fileExists,
+        'tsconfig.json'
+    );
+
+    if (!configPath) {
```

Comment:
I'm not sure we should force a tsconfig.json

Reply:
we are just using exec sync now